### PR TITLE
npe2 docs (ref #3769)

### PR DESCRIPTION
**Pull request by nclack**
# Context

This PR is a rollup after feedback on #3769.  Most (but not all) conversations in that PR have been resolved. Please add further review to this one.  I imagine this represents a first public offering of the docs. There are changes I'd like to make in the future though, including:

- [ ] making the manifest spec more maintainable
- [ ] separating the narrative content in the manifest spec
- [ ] creating focused guides for each kind of plugin

I'll add to the above list as I get comments, and will consolidate/break those out into follow-up issues later.

# Description

This adds documentation for npe2 to the napari docs.  Some of the details are awaiting the last few changes to the npe2 manifest before release.

~I'll keep this as draft until npe2 release. Please feel free to review while this is in draft.~

There is a 

- [x] npe2 manifest specification
- [x] npe2 getting started guide
- [x] npe2 migration guide
- [x] introduction of npe2 on the plugin index page
- [x] added new pages to the top level _toc.yml

## Type of change
- [x] This change requires a documentation update

## Todo
- [x] update for napari/npe2#51
- [x] update for napari/npe2#49
- [x] add announcement date it "introducing npe2" news item. Gives it some time context.
- [x] add "introducing npe2" to various pages (not just the plugin index page)
- [x] hint that napari-plugin-engine will be deprecated soon (at the support/doc level).

### CODE REVIEW

File: docs/plugins/npe2_manifest_specification.md
Comment (andy-sweet) [inline docs/plugins/npe2_manifest_specification.md]: I'm just browsing for now (no full review yet!) here, but my built-in typo sensor picked this one out.
```suggestion
    - display_name: Tabueran Kiribati
```

File: docs/plugins/npe2_manifest_specification.md
Comment (andy-sweet) [inline docs/plugins/npe2_manifest_specification.md]: Just a warning about this name. `Layer.metadata` is an existing public property that refers to a different `dict` than this one, which comes from `Layer.as_layer_data_tuple` (which happens to include `Layer.metadata`).

At a minimum in the docs, it would be good to add a general note/warning about this name conflict. Ideally we would resolve the conflict by renaming one of these two (I'm not sure `meta` and `metadata` is enough of a distinction), but this isn't the PR for that.

Comment (nclack) [reply docs/plugins/npe2_manifest_specification.md]: Would this make it clearer?
```suggestion
: Layer metadata specified as an unstructured `dict`
```

Comment (andy-sweet) [reply docs/plugins/npe2_manifest_specification.md]: Not really, because `Layer.metadata` is also an unstructured dict: https://github.com/napari/napari/blob/fcde7328ac6a561998c1cacaca9dc76492c81b08/napari/layers/base/base.py#L73

Comment (andy-sweet) [reply docs/plugins/npe2_manifest_specification.md]: I'm not sure what to suggest here and it's not something we should spend too much time on right now, so unless someone else feels strongly here, we can just ignore this for now.

File: docs/plugins/npe2_getting_started.md
Comment (sofroniewn) [inline docs/plugins/npe2_getting_started.md:16]: Maybe add something like "if you want to convert and existing plugin please see other document (link)"

**Review by sofroniewn — APPROVED** [review]
This is truly incredible @nclack! Great work on the docs, and thanks so much all who have reviewed @andy-sweet @chili-chiu @DragaDoncila! I only had one trivial comment that can be added later. I think this looks good enough to merge now. I'd like to get this in tomorrow morning, just before the plugin accelerator meeting. @tlambert03 @nclack what are your thoughts? We can always continue to improve after merge, but I think it will make a big difference to have this in

**Comment by sofroniewn** [comment]
Just saw:

> I'll keep this as draft until npe2 release. Please feel free to review while this is in draft.

I'm not sure if that changes timing. I'll wait for further instruction, but docs look good to me

**Comment by nclack** [comment]
> Just saw:
> 
> > I'll keep this as draft until npe2 release. Please feel free to review while this is in draft.
> 
> I'm not sure if that changes timing. I'll wait for further instruction, but docs look good to me

These are good to go!

**Comment by sofroniewn** [comment]
> These are good to go!

Ok nice, I will merge first thing tomorrow morning when I wake up then, unless i hear otherwise!!

File: docs/plugins/npe2_manifest_specification.md
Comment (tlambert03) [reply docs/plugins/npe2_manifest_specification.md]: Right; the problem was the choice of the parameter name "meta".  So when we describe it, we should call it "layer attributes"

Comment (nclack) [reply docs/plugins/npe2_manifest_specification.md]: changed description to 'layer attributes'

File: docs/plugins/npe2_getting_started.md
Comment (nclack) [reply docs/plugins/npe2_getting_started.md:16]: added to the top level admonition on that page.

**Comment by sofroniewn** [comment]
We've got some fails on the bundle, but otherwise looks good. Will merge

**Comment by tlambert03** [comment]
> We can always continue to improve after merge

let's do this! and soon :)  as discussed with @nclack yesterday, I think there's a lot of work to be done here.  Let's not let the merge soothe us into feeling like it's off the plate :)

**Comment by nclack** [comment]
> there's a lot of work to be done here

absolutely. hopefully, this pr creates a baseline we can build smaller changes from.

**Pull request merged**
